termutil/is_tty.go

```diff
@@ -0,0 +1,11 @@
+package termutil
+
+import "os"
+
+// IsTTY returns true if a terminal is detected
+func IsTTY() bool {
+	if fileInfo, _ := os.Stdout.Stat(); (fileInfo.Mode() & os.ModeCharDevice) != 0 {
+		return true
+	}
+	return false
+}
```

Comment:
BTW, I used something like this here: https://github.com/earthly/earthly/blob/main/builder/solver_monitor.go#L81